src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java

```diff
@@ -105,6 +108,39 @@ void surrogatesWithRaw() throws Exception
         jp.close();
     }
 
+    @Test
+    void lastSegmentCharSplitSurrogateCharInTwoSegments() throws Exception
+    {
+        // segments split every 1000 chars.
+        // We need a string with length 1001 where the surrogate is
+        // at 1000 and 1001 positions
+        int count = 999;
+        char[] chars = new char[count];
+        java.util.Arrays.fill(chars, 'x');
+        String base = new String(chars);
+
+        final String VALUE = base + "\uD83E\uDEE1";
+
+        // alas, we have to pull the recycler directly here...
+        final BufferRecycler br = JSON_F._getBufferRecycler();
+        ByteArrayBuilder bb = new ByteArrayBuilder(br);
+        JsonGenerator g = JSON_F.createGenerator(bb);
+        g.enable(JsonGenerator.Feature.COMBINE_UNICODE_SURROGATES_IN_UTF8);
+
+        g.writeStartArray();
+        g.writeString(VALUE);
+        g.writeEndArray();
+
+        g.close();
+
+        String result = new String(bb.toByteArray());
```

Comment:
Could you specify StandardCharsets.UTF_8 so that we are not reliant on the default encoding (that can differ depending on OS, etc.?

Reply:
Added